plugins/search-react/src/components/SearchFilter/SearchFilter.tsx

```diff
@@ -140,27 +154,33 @@ export const CheckboxFilter = (props: SearchFilterComponentProps) => {
 /**
  * @public
  */
-export const SelectFilter = (props: SearchFilterComponentProps) => {
+export function SelectFilter<T extends JsonValue = string>(
```

Comment:
hm, good point, made them all `string | string[]`.
Not sure about last point, but having generic type allows you to specify type exactly on client side, so e.g. for multiselect component we will specify explicitly `string[]` and e.g. on `renderValue` will get correct type hint

Reply:
I guess the JsonValues type give us the opportunity to support values such as numbers for example, and I think we could have a situation where a select component with numbers is used. Lets imagine a select filter with number of results for example or version or something like that.

Reply:
I see there few points:
1) JsonValue is too generic thing, it includes objects for example as well which I think should not be here
2) such changes in types will cause a bunch of other changes, e.g. to prove that if you use `number` there will be actually number, because dom makes everything string and also all code dependencies because for now codebase considers only 2 options: `string | string[]`